{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0fda6169",
   "metadata": {},
   "outputs": [
    {
     "ename": "ConfigurationError",
     "evalue": "The resolution lifetime expired after 21.153 seconds: Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mLifetimeTimeout\u001b[0m                           Traceback (most recent call last)",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36m_resolve_uri\u001b[0;34m(self, encapsulate_errors)\u001b[0m\n\u001b[1;32m     88\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 89\u001b[0;31m             results = _resolve(\n\u001b[0m\u001b[1;32m     90\u001b[0m                 \u001b[0;34m\"_\"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__srv\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\"._tcp.\"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__fqdn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"SRV\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlifetime\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__connect_timeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36m_resolve\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     42\u001b[0m         \u001b[0;31m# dnspython >= 2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 43\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mresolver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mresolve\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     44\u001b[0m     \u001b[0;31m# dnspython 1.X\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/dns/resolver.py\u001b[0m in \u001b[0;36mresolve\u001b[0;34m(qname, rdtype, rdclass, tcp, source, raise_on_no_answer, source_port, lifetime, search)\u001b[0m\n\u001b[1;32m   1192\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1193\u001b[0;31m     return get_default_resolver().resolve(qname, rdtype, rdclass, tcp, source,\n\u001b[0m\u001b[1;32m   1194\u001b[0m                                           \u001b[0mraise_on_no_answer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msource_port\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/dns/resolver.py\u001b[0m in \u001b[0;36mresolve\u001b[0;34m(self, qname, rdtype, rdclass, tcp, source, raise_on_no_answer, source_port, lifetime, search)\u001b[0m\n\u001b[1;32m   1065\u001b[0m                     \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msleep\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbackoff\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1066\u001b[0;31m                 timeout = self._compute_timeout(start, lifetime,\n\u001b[0m\u001b[1;32m   1067\u001b[0m                                                 resolution.errors)\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/dns/resolver.py\u001b[0m in \u001b[0;36m_compute_timeout\u001b[0;34m(self, start, lifetime, errors)\u001b[0m\n\u001b[1;32m    878\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mduration\u001b[0m \u001b[0;34m>=\u001b[0m \u001b[0mlifetime\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 879\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mLifetimeTimeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mduration\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merrors\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merrors\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    880\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mmin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlifetime\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mduration\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mLifetimeTimeout\u001b[0m: The resolution lifetime expired after 21.153 seconds: Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mConfigurationError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/c2/x_cgj8b10s38q85qqx_c88hr0000gn/T/ipykernel_32333/260775975.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mpymongo\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmongo_client\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mMongoClient\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mData\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mMongoClient\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"mongodb+srv://chethanreddy1234:chethan1234@cluster0.yvbx0ko.mongodb.net/?retryWrites=true&w=majority\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mData\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/mongo_client.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, host, port, document_class, tz_aware, connect, type_registry, **kwargs)\u001b[0m\n\u001b[1;32m    734\u001b[0m                         \u001b[0mkeyword_opts\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcased_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"connecttimeoutms\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    735\u001b[0m                     )\n\u001b[0;32m--> 736\u001b[0;31m                 res = uri_parser.parse_uri(\n\u001b[0m\u001b[1;32m    737\u001b[0m                     \u001b[0mentity\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    738\u001b[0m                     \u001b[0mport\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/uri_parser.py\u001b[0m in \u001b[0;36mparse_uri\u001b[0;34m(uri, default_port, validate, warn, normalize, connect_timeout, srv_service_name, srv_max_hosts)\u001b[0m\n\u001b[1;32m    540\u001b[0m         \u001b[0mconnect_timeout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconnect_timeout\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0moptions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"connectTimeoutMS\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    541\u001b[0m         \u001b[0mdns_resolver\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_SrvResolver\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfqdn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconnect_timeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msrv_service_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msrv_max_hosts\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 542\u001b[0;31m         \u001b[0mnodes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdns_resolver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_hosts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    543\u001b[0m         \u001b[0mdns_options\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdns_resolver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_options\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    544\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mdns_options\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36mget_hosts\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    119\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    120\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget_hosts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 121\u001b[0;31m         \u001b[0m_\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnodes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_srv_response_and_hosts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    122\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mnodes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    123\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36m_get_srv_response_and_hosts\u001b[0;34m(self, encapsulate_errors)\u001b[0m\n\u001b[1;32m     99\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    100\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_get_srv_response_and_hosts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencapsulate_errors\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 101\u001b[0;31m         \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_resolve_uri\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mencapsulate_errors\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    102\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    103\u001b[0m         \u001b[0;31m# Construct address tuples\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36m_resolve_uri\u001b[0;34m(self, encapsulate_errors)\u001b[0m\n\u001b[1;32m     95\u001b[0m                 \u001b[0;32mraise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     96\u001b[0m             \u001b[0;31m# Else, raise all errors as ConfigurationError.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 97\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mConfigurationError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     98\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresults\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     99\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mConfigurationError\u001b[0m: The resolution lifetime expired after 21.153 seconds: Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out."
     ]
    }
   ],
   "source": [
    "from pymongo.mongo_client import MongoClient\n",
    "\n",
    "Data = MongoClient(\"mongodb+srv://chethanreddy1234:chethan1234@cluster0.yvbx0ko.mongodb.net/?retryWrites=true&w=majority\")\n",
    "print(Data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "defac236",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collection(Database(MongoClient(host=['ac-s9dsrxv-shard-00-00.yvbx0ko.mongodb.net:27017', 'ac-s9dsrxv-shard-00-02.yvbx0ko.mongodb.net:27017', 'ac-s9dsrxv-shard-00-01.yvbx0ko.mongodb.net:27017'], document_class=dict, tz_aware=False, connect=True, retrywrites=True, w='majority', authsource='admin', replicaset='atlas-1ohy5i-shard-0', tls=True), 'Test'), 'Test')\n"
     ]
    }
   ],
   "source": [
    "ConnectData = Data['Test']['Test']\n",
    "print(ConnectData)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3b1fe662",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.results.InsertOneResult at 0x7fac018c17c0>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ConnectData.insert_one({\"Name\" : \"VIT\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "370bd403",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.results.DeleteResult at 0x7f95691fca00>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ConnectData.delete_many({})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "03a133b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': ObjectId('63a19a5931f5f08153aa2a6f'),\n",
       " 'Patient_Name': 'Rev',\n",
       " 'Patient_Age': '33',\n",
       " 'Gender': 'Female'}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ConnectData.find_one({\"Patient_Name\" : \"Rev\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b04944ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['$2b$12$KujTAegFHLK5bLTHAP9tLeXElnSt934IsGHr8gc8HaZSreDdkJ9r6', '$2b$12$/22vdFs1W4WsUzvHRNxMqOChOtLmVW1ouTb/uAx6UJZurAHvLR5x.']\n"
     ]
    }
   ],
   "source": [
    "import streamlit_authenticator as stauth\n",
    "\n",
    "hashed_passwords = stauth.Hasher(['123', '456']).generate()\n",
    "\n",
    "print(hashed_passwords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8b663f0f",
   "metadata": {},
   "outputs": [
    {
     "ename": "ConfigurationError",
     "evalue": "The resolution lifetime expired after 21.150 seconds: Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mLifetimeTimeout\u001b[0m                           Traceback (most recent call last)",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36m_resolve_uri\u001b[0;34m(self, encapsulate_errors)\u001b[0m\n\u001b[1;32m     88\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 89\u001b[0;31m             results = _resolve(\n\u001b[0m\u001b[1;32m     90\u001b[0m                 \u001b[0;34m\"_\"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__srv\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\"._tcp.\"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__fqdn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"SRV\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlifetime\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__connect_timeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36m_resolve\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     42\u001b[0m         \u001b[0;31m# dnspython >= 2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 43\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mresolver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mresolve\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     44\u001b[0m     \u001b[0;31m# dnspython 1.X\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/dns/resolver.py\u001b[0m in \u001b[0;36mresolve\u001b[0;34m(qname, rdtype, rdclass, tcp, source, raise_on_no_answer, source_port, lifetime, search)\u001b[0m\n\u001b[1;32m   1192\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1193\u001b[0;31m     return get_default_resolver().resolve(qname, rdtype, rdclass, tcp, source,\n\u001b[0m\u001b[1;32m   1194\u001b[0m                                           \u001b[0mraise_on_no_answer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msource_port\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/dns/resolver.py\u001b[0m in \u001b[0;36mresolve\u001b[0;34m(self, qname, rdtype, rdclass, tcp, source, raise_on_no_answer, source_port, lifetime, search)\u001b[0m\n\u001b[1;32m   1065\u001b[0m                     \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msleep\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbackoff\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1066\u001b[0;31m                 timeout = self._compute_timeout(start, lifetime,\n\u001b[0m\u001b[1;32m   1067\u001b[0m                                                 resolution.errors)\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/dns/resolver.py\u001b[0m in \u001b[0;36m_compute_timeout\u001b[0;34m(self, start, lifetime, errors)\u001b[0m\n\u001b[1;32m    878\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mduration\u001b[0m \u001b[0;34m>=\u001b[0m \u001b[0mlifetime\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 879\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mLifetimeTimeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mduration\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merrors\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merrors\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    880\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mmin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlifetime\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mduration\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mLifetimeTimeout\u001b[0m: The resolution lifetime expired after 21.150 seconds: Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mConfigurationError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/c2/x_cgj8b10s38q85qqx_c88hr0000gn/T/ipykernel_32333/4084827029.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mData\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mMongoClient\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"mongodb+srv://chethanreddy1234:chethan1234@cluster0.yvbx0ko.mongodb.net/?retryWrites=true&w=majority\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mConnectData\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mData\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Test'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Test'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/mongo_client.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, host, port, document_class, tz_aware, connect, type_registry, **kwargs)\u001b[0m\n\u001b[1;32m    734\u001b[0m                         \u001b[0mkeyword_opts\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcased_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"connecttimeoutms\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    735\u001b[0m                     )\n\u001b[0;32m--> 736\u001b[0;31m                 res = uri_parser.parse_uri(\n\u001b[0m\u001b[1;32m    737\u001b[0m                     \u001b[0mentity\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    738\u001b[0m                     \u001b[0mport\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/uri_parser.py\u001b[0m in \u001b[0;36mparse_uri\u001b[0;34m(uri, default_port, validate, warn, normalize, connect_timeout, srv_service_name, srv_max_hosts)\u001b[0m\n\u001b[1;32m    540\u001b[0m         \u001b[0mconnect_timeout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconnect_timeout\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0moptions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"connectTimeoutMS\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    541\u001b[0m         \u001b[0mdns_resolver\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_SrvResolver\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfqdn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconnect_timeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msrv_service_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msrv_max_hosts\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 542\u001b[0;31m         \u001b[0mnodes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdns_resolver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_hosts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    543\u001b[0m         \u001b[0mdns_options\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdns_resolver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_options\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    544\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mdns_options\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36mget_hosts\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    119\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    120\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget_hosts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 121\u001b[0;31m         \u001b[0m_\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnodes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_srv_response_and_hosts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    122\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mnodes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    123\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36m_get_srv_response_and_hosts\u001b[0;34m(self, encapsulate_errors)\u001b[0m\n\u001b[1;32m     99\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    100\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_get_srv_response_and_hosts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencapsulate_errors\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 101\u001b[0;31m         \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_resolve_uri\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mencapsulate_errors\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    102\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    103\u001b[0m         \u001b[0;31m# Construct address tuples\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Applications/anaconda3/lib/python3.9/site-packages/pymongo/srv_resolver.py\u001b[0m in \u001b[0;36m_resolve_uri\u001b[0;34m(self, encapsulate_errors)\u001b[0m\n\u001b[1;32m     95\u001b[0m                 \u001b[0;32mraise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     96\u001b[0m             \u001b[0;31m# Else, raise all errors as ConfigurationError.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 97\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mConfigurationError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     98\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresults\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     99\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mConfigurationError\u001b[0m: The resolution lifetime expired after 21.150 seconds: Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out.; Server 192.168.152.46 UDP port 53 answered The DNS operation timed out."
     ]
    }
   ],
   "source": [
    "Data = MongoClient(\"mongodb+srv://chethanreddy1234:chethan1234@cluster0.yvbx0ko.mongodb.net/?retryWrites=true&w=majority\")\n",
    "ConnectData = Data['Test']['Test']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fc0a5524",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'_id': ObjectId('63a43daf886574d1c4aebfd3'), 'Patient_Id': 'ST391', 'Patient_Name': 'Mark Bid', 'Patient_Age': '34', 'Gender': 'Male', 'contact_number': '4567568576', 'Occupation': 'Teacher', 'Address': '3465 Vellore', 'DateOfAsses': '2022-12-22', 'Complaint': 'Heavy Fever', 'Past_History': 'None', 'Surgical_History': 'None', 'Referal_Doctor': 'Jeff Bezos', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n"
     ]
    }
   ],
   "source": [
    "Results = ConnectData.find({\"Patient_Id\" : \"ST391\"})\n",
    "for i in Results:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "128b1bfe",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pymongo.mongo_client import MongoClient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "58ae0e5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "Data = MongoClient(\"mongodb://chethanreddy1234:chethan1234@ac-s9dsrxv-shard-00-00.yvbx0ko.mongodb.net:27017,ac-s9dsrxv-shard-00-01.yvbx0ko.mongodb.net:27017,ac-s9dsrxv-shard-00-02.yvbx0ko.mongodb.net:27017/?ssl=true&replicaSet=atlas-1ohy5i-shard-0&authSource=admin&retryWrites=true&w=majority\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ad0b7012",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'_id': ObjectId('63a435b6dcd752be018ef23f'), 'Patient_Id': 'ST309', 'Patient_Name': 'Jack Henry', 'Patient_Age': '24', 'Gender': 'Male', 'contact_number': '564574657456', 'Occupation': 'SDE', 'Address': '325-534 Riakc , ad', 'DateOfAsses': '2022-12-22', 'Complaint': 'Fever', 'Past_History': 'None', 'Surgical_History': 'None', 'Referal_Doctor': 'Elon Musk', 'Patient_Documents': []}\n",
      "{'_id': ObjectId('63a43656dcd752be018ef24a'), 'Patient_Id': 'ST47', 'Patient_Name': 'Henry  Mark', 'Patient_Age': '24', 'Gender': 'Male', 'contact_number': '7875848438', 'Occupation': 'Software Developer', 'Address': '20-3-5-6 Katpadi Vellore', 'DateOfAsses': '2022-12-22', 'Complaint': 'Headache', 'Past_History': 'None', 'Surgical_History': 'None', 'Referal_Doctor': 'Albert', 'Patient_Documents': []}\n",
      "{'_id': ObjectId('63a43c2d886574d1c4aebfc7'), 'Patient_Id': 'ST538', 'Patient_Name': 'Henry Jack', 'Patient_Age': '34', 'Gender': 'Male', 'contact_number': '645675676', 'Occupation': 'Data Science', 'Address': '34534ktjfkljad', 'DateOfAsses': '2022-12-22', 'Complaint': 'Fever,Cold', 'Past_History': 'None', 'Surgical_History': 'NOne', 'Referal_Doctor': 'elon', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n",
      "{'_id': ObjectId('63a43daf886574d1c4aebfd3'), 'Patient_Id': 'ST391', 'Patient_Name': 'Mark Bid', 'Patient_Age': '34', 'Gender': 'Male', 'contact_number': '4567568576', 'Occupation': 'Teacher', 'Address': '3465 Vellore', 'DateOfAsses': '2022-12-22', 'Complaint': 'Heavy Fever', 'Past_History': 'None', 'Surgical_History': 'None', 'Referal_Doctor': 'Jeff Bezos', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n",
      "{'_id': ObjectId('63a47844381c56c43827f32e'), 'Patient_Id': 'ST641', 'Patient_Name': ';alskdjf', 'Patient_Age': '34', 'Gender': 'Male', 'contact_number': '3435464657', 'Occupation': 'sfdkljg', 'Address': 's;lfkdgjsl', 'DateOfAsses': '2022-12-22', 'Complaint': ';fkjg', 'Past_History': 'slkfgjs', 'Surgical_History': 'lkfdjg', 'Referal_Doctor': 'lskfdjg', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n",
      "{'_id': ObjectId('63a50b9f4080a24bb4e63e2b'), 'Patient_Id': 'ST545', 'Patient_Name': 'kldjf3', 'Patient_Age': '44', 'Gender': 'Male', 'contact_number': '34543543534kl', 'Occupation': 'jk', 'Address': 'kljhkl', 'DateOfAsses': '2022-12-23', 'Complaint': 'lkhjlkjhlj', 'Past_History': 'lkj;lkj', 'Surgical_History': 'lkhlkjh', 'Referal_Doctor': 'kjhlkjh', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n",
      "{'_id': ObjectId('63a50ba44080a24bb4e63e2d'), 'Patient_Id': 'ST388', 'Patient_Name': 'kldjf3', 'Patient_Age': '44', 'Gender': 'Male', 'contact_number': '34543543534kl', 'Occupation': 'jk', 'Address': 'kljhkl', 'DateOfAsses': '2022-12-23', 'Complaint': 'lkhjlkjhlj', 'Past_History': 'lkj;lkj', 'Surgical_History': 'lkhlkjh', 'Referal_Doctor': 'kjhlkjh', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n",
      "{'_id': ObjectId('63a50ba84080a24bb4e63e30'), 'Patient_Id': 'ST970', 'Patient_Name': 'kldjf3', 'Patient_Age': '44', 'Gender': 'Male', 'contact_number': '34543543534kl', 'Occupation': 'jk', 'Address': 'kljhkl', 'DateOfAsses': '2022-12-23', 'Complaint': 'lkhjlkjhlj', 'Past_History': 'lkj;lkj', 'Surgical_History': 'lkhlkjh', 'Referal_Doctor': 'kjhlkjh', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n",
      "{'_id': ObjectId('63a50bb34080a24bb4e63e32'), 'Patient_Id': 'ST945', 'Patient_Name': 'kldjf3', 'Patient_Age': '44', 'Gender': 'Male', 'contact_number': '34543543534kl', 'Occupation': 'jk', 'Address': 'kljhkl', 'DateOfAsses': '2022-12-23', 'Complaint': 'lkhjlkjhlj', 'Past_History': 'lkj;lkj', 'Surgical_History': 'lkhlkjh', 'Referal_Doctor': 'kjhlkjh', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n",
      "{'_id': ObjectId('63a50c1e4080a24bb4e63e34'), 'Patient_Id': 'ST101', 'Patient_Name': 'kldjf3', 'Patient_Age': '44', 'Gender': 'Male', 'contact_number': '34543543534kl', 'Occupation': 'jk', 'Address': 'kljhkl', 'DateOfAsses': '2022-12-23', 'Complaint': 'lkhjlkjhlj', 'Past_History': 'lkj;lkj', 'Surgical_History': 'lkhlkjh', 'Referal_Doctor': 'kjhlkjh', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n",
      "{'_id': ObjectId('63a50c364080a24bb4e63e38'), 'Patient_Id': 'ST56', 'Patient_Name': 'fsdgs', 'Patient_Age': 'fggsf', 'Gender': 'Male', 'contact_number': 'fdgsf', 'Occupation': 'g', 'Address': 'fdsgfs', 'DateOfAsses': '2022-12-23', 'Complaint': 'fgsf', 'Past_History': 'sfgsf', 'Surgical_History': 'sfgg', 'Referal_Doctor': 'fdgsf', 'Patient_Documents': [], 'Senior_Doctors_Prescription': [], 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n"
     ]
    }
   ],
   "source": [
    "ConnectData = Data[\"Test\"][\"Test\"]\n",
    "Result = ConnectData.find({})\n",
    "\n",
    "for i in Result:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "b232f14d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.results.UpdateResult at 0x7fda6a75b220>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ConnectData.update_one(\n",
    "        {\"Patient_Id\":\"ST56\"},\n",
    "        {\n",
    "                \"$set\":{\n",
    "                        \"Senior_Doctors_Prescription\": {\"Jack\" : 1234}\n",
    "                        }\n",
    "        }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "07317b1f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'_id': ObjectId('63a50c364080a24bb4e63e38'), 'Patient_Id': 'ST56', 'Patient_Name': 'fsdgs', 'Patient_Age': 'fggsf', 'Gender': 'Male', 'contact_number': 'fdgsf', 'Occupation': 'g', 'Address': 'fdsgfs', 'DateOfAsses': '2022-12-23', 'Complaint': 'fgsf', 'Past_History': 'sfgsf', 'Surgical_History': 'sfgg', 'Referal_Doctor': 'fdgsf', 'Patient_Documents': [], 'Senior_Doctors_Prescription': {'S_Diagosis': 'kadlsh', 'Treatment': 'jklh', 'Exercise': 'kjh', 'review': 'jhgjhg', 'Home_Advice': 'kjhjh', 'Up_Treatment': 'kjhlkjh', 'MRI_Impression': 'kjhkjh'}, 'Junior_Doctors_Prescription': [], 'Trainer_Prescription': []}\n"
     ]
    }
   ],
   "source": [
    "Result = ConnectData.find({'Patient_Id': 'ST56'})\n",
    "\n",
    "for i in Result:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "943870f3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6 (default, Sep 26 2022, 11:37:49) \n[Clang 14.0.0 (clang-1400.0.29.202)]"
  },
  "vscode": {
   "interpreter": {
    "hash": "31f2aee4e71d21fbe5cf8b01ff0e069b9275f58929596ceb00d14d90e3e16cd6"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
